nixos/modules/programs/nh.nix

```
@@ -77,7 +77,7 @@ in
     environment = lib.mkIf cfg.enable {
       systemPackages = [ cfg.package ];
       variables = lib.mkIf (cfg.flake != null) {
-        FLAKE = cfg.flake;
+        NH_FLAKE = cfg.flake;
```

would be better to use something similar to what home-manager does in case someone wants to use an old nh version
```suggestion
        let
          packageVersion = lib.getVersion cfg.package;
        in
        if lib.versionAtLeast packageVersion "4.0.0" then { NH_FLAKE = cfg.flake; } else { FLAKE = cfg.flake; }
```

---

The unstable branch already has the latest version, so people using this change should have a newer nh. Idk how this should be the case, but I don't think it's worth it anyways.

---

In general, HM will make a scene if the nixpkgs/nixos version is on a different branch: https://github.com/nix-community/home-manager/blob/2f5819a962489e037a57835f63ed6ff8dbc2d5fb/modules/home-environment.nix#L539

So some guardrails will not hurt here but I severely doubt this will ever be a problem to someone.